src/Path/REPL.hs

```diff
@@ -202,23 +205,24 @@ script packageSources = evalState (ModuleGraph mempty :: ModuleGraph QName (Term
         reload = do
           put (Resolution mempty)
           let n = length packageSources
-          sorted <- traverse (parseFile . whole . module' <*> id) packageSources >>= loadOrder . moduleGraph >>= traverse resolveModule
+          sorted <- traverse (parseFile . whole . module' <*> id) packageSources >>= loadOrder . moduleGraph
 
-          runDeps . for_ (zip [(1 :: Int)..] sorted) $ \ (i, m) -> skipDeps m $ do
+          checked <- runDeps . for (zip [(1 :: Int)..] sorted) $ \ (i, m) -> skipDeps m $ do
             let name    = moduleName m
                 ordinal = brackets (pretty i <+> pretty "of" <+> pretty n)
                 path    = parens (pretty (modulePath m))
             print (ordinal <+> pretty "Compiling" <+> pretty name <+> path)
             table <- get
-            (errs, res) <- runState [] (runReader (table :: ModuleTable) (elabModule m))
+            (errs, (ctx, (env, res))) <- runState [] (runReader (table :: ModuleTable) (runFresh (runState Context.empty (runState Env.empty (resolveModule m >>= traverse desugar >>= elabModule)))))
             if Prelude.null errs then
-              modify (Map.insert name res)
+              modify (Map.insert name (ctx, env))
             else do
               for_ errs printElabError
               modify (name:)
-          put (moduleGraph sorted)
+            pure (Just res)
+          put (moduleGraph (catMaybes checked))
```

Comment:
We store the graph of elaborated modules instead of the initial graph. This means that we won’t be able to look up docs for declarations that didn’t typecheck, unfortunately.